lms/services/lti_outcomes.py

```diff
@@ -78,25 +72,15 @@ def record_result(  # pylint:disable=no-self-use
                 "resultScore": {"language": "en", "textString": score}
             }
 
-        # Canvas specific adaptations
-        self._canvas_request_modification(request, **canvas_extras)
-
-        self._send_request({"replaceResultRequest": request})
-
-    @classmethod
-    def _canvas_request_modification(
-        cls, request, lti_launch_url=None, submitted_at=None, **_
-    ):
-        # For details of Canvas extensions see:
-        # https://erau.instructure.com/doc/api/file.assignment_tools.html
+        if pre_record_hook:
+            request = pre_record_hook(score=score, request_body=request)
 
-        if lti_launch_url:
-            request["resultRecord"].setdefault("result", {})["resultData"] = {
-                "ltiLaunchUrl": lti_launch_url
-            }
+            if not isinstance(request, dict):
+                raise TypeError(
+                    "The pre-record hook must return the request body as a dict"
+                )
```

Comment:
A little defensive programming is probably in order here. If we end up with lots of hooks we might want a way of calling them that catches argument errors and expects a particular type. Possible raising `BadHookImplementation`, `BadHookReturnType` or something similar.